sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/http/HttpEndpointMethodDefinition.scala

```diff
@@ -0,0 +1,411 @@
+
+    def convertHttpRule(jHttpRule: JavaHttpRule): Option[HttpRule] = {
+      try {
+        val httpRule = HttpRule.of(
+          selector = jHttpRule.getSelector,
+          pattern = {
+            jHttpRule.getPatternCase match {
+              case PatternCase.GET             => HttpRule.Pattern.Get(jHttpRule.getGet)
+              case PatternCase.POST            => HttpRule.Pattern.Post(jHttpRule.getPost)
+              case PatternCase.PUT             => HttpRule.Pattern.Put(jHttpRule.getPut)
+              case PatternCase.PATCH           => HttpRule.Pattern.Patch(jHttpRule.getPatch)
+              case PatternCase.DELETE          => HttpRule.Pattern.Delete(jHttpRule.getDelete)
+              case PatternCase.PATTERN_NOT_SET => HttpRule.Pattern.Empty
+              case PatternCase.CUSTOM =>
+                HttpRule.Pattern.Custom(
+                  CustomHttpPattern(kind = jHttpRule.getCustom.getKind, path = jHttpRule.getCustom.getPath))
+            }
+          },
+          body = jHttpRule.getBody,
+          responseBody = jHttpRule.getResponseBody,
+          additionalBindings = jHttpRule.getAdditionalBindingsList.asScala
+            .flatMap(convertHttpRule)
+            .toSeq // FIXME can we make this tailrec?
+        )
+        Some(httpRule)
+      } catch {
+        case NonFatal(e) =>
+          log.error(s"Failed to convert $jHttpRule", e)
+          None
+      }
+    }
+
+    def convertMethodOptions(method: MethodDescriptor): spbMethodOptions =
+      spbMethodOptions
+        .fromJavaProto(method.getOptions)
+        .withExtension(AnnotationsProto.http)(
+          convertHttpRule(method.getOptions.getExtension(JavaAnnotationsProto.http)))
+
+    val rule = convertMethodOptions(method).extension(AnnotationsProto.http) match {
+      case Some(rule) =>
+        log.info(s"Using configured HTTP API endpoint using [$rule]")
+        rule
+      case None =>
+        val rule = HttpRule.of(
+          selector = method.getFullName, // We know what thing we are proxying
+          body = "*", // Parse all input
+          responseBody = "", // Include all output
+          additionalBindings = Nil, // No need for additional bindings
+          pattern = HttpRule.Pattern.Post((Path / service.getFullName / method.getName).toString))
+        log.info(s"Using generated HTTP API endpoint using [$rule]")
+        rule
+    }
+    rule +: rule.additionalBindings
+  }
+
+  // This method validates the configuration and returns values obtained by parsing the configuration
+  private final def extractAndValidate(methDesc: MethodDescriptor, rule: HttpRule)
+      : (HttpMethod, PathTemplateParser.ParsedTemplate, ExtractPathParameters, Descriptor, Option[FieldDescriptor]) = {
+    // Validate selector
+    if (rule.selector != "" && rule.selector != methDesc.getFullName)
+      throw new Exception(
+        "HttpApi.invalidSelector(rule.selector, methDesc)"
+      ) // FIXME fix all the exceptions to useful outputs
```

Comment:
@octonato while addressing your comments I noticed there was something else I also forgot to update... if you have a look on this file there's a bunch of exceptions being thrown. In the original proxy version, these make use of a `HttpApi` defining the types of errors. We don't have such errors mapped on the SDK side, AFAIK. I'm thinking of just adding a shared wrapper exception (maybe `HttpEndpointParsingException`) with a custom message. Let me know if you think there might be a better option.